File: iOS/FolioApp/ViewModels/HomeViewModel.swift
import Foundation

@MainActor
class HomeViewModel: ObservableObject {
    @Published var trending: [Project] = []
    @Published var latest: [Project] = []
    @Published var categories: [String] = [
        "Графический дизайн", "UI/UX", "Иллюстрация", "3D",
        "Моушн-дизайн", "Фотография", "Веб-дизайн", "Брендинг"
    ]
    @Published var isLoading = false
    @Published var error: String?

    private let api = APIService.shared

    func load() async {
        isLoading = true
        do {
            async let trendingReq: ProjectsResponse = api.get("/projects", query: ["sort": "popular", "limit": "8"], auth: false)
            async let latestReq: ProjectsResponse = api.get("/projects", query: ["sort": "latest", "limit": "8"], auth: false)
            let (t, l) = try await (trendingReq, latestReq)
            trending = t.projects
            latest = l.projects
        } catch let err as APIError {
            error = err.errorDescription
        } catch {
            self.error = error.localizedDescription
        }
        isLoading = false
    }
}

File: Android/app/src/main/java/com/folio/app/ui/home/HomeViewModel.kt
package com.folio.app.ui.home

import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.folio.app.data.api.ApiService
import com.folio.app.data.models.Project
import dagger.hilt.android.lifecycle.HiltViewModel
import kotlinx.coroutines.async
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow
import kotlinx.coroutines.launch
import javax.inject.Inject

@HiltViewModel
class HomeViewModel @Inject constructor(
    private val api: ApiService
) : ViewModel() {

    private val _trending = MutableStateFlow<List<Project>>(emptyList())
    val trending: StateFlow<List<Project>> = _trending

    private val _latest = MutableStateFlow<List<Project>>(emptyList())
    val latest: StateFlow<List<Project>> = _latest

    private val _isLoading = MutableStateFlow(false)
    val isLoading: StateFlow<Boolean> = _isLoading

    val categories = listOf(
        "Графический дизайн", "UI/UX", "Иллюстрация", "3D",
        "Моушн-дизайн", "Фотография", "Веб-дизайн", "Брендинг"
    )

    fun load() {
        viewModelScope.launch {
            _isLoading.value = true
            try {
                val trendingDeferred = async {
                    api.getProjects(mapOf("sort" to "popular", "limit" to "8"))
                }
                val latestDeferred = async {
                    api.getProjects(mapOf("sort" to "latest", "limit" to "8"))
                }
                trendingDeferred.await().body()?.projects?.let { _trending.value = it }
                latestDeferred.await().body()?.projects?.let { _latest.value = it }
            } catch (_: Exception) {}
            _isLoading.value = false
        }
    }
}
